### PR TITLE
SLLS-98

### PR DESCRIPTION
**Pull request by kirill-knize-sonarsource**

### CODE REVIEW

File: src/main/java/org/sonarsource/sonarlint/ls/SonarLintExtendedLanguageServer.java [hunk 1]
```diff
@@ -214,4 +214,51 @@ public void setConnectionId(String connectionId) {
 
   @JsonRequest("sonarlint/getRemoteProjectNames")
   CompletableFuture<Map<String, String>> getRemoteProjectNames(GetRemoteProjectsNamesParams params);
+
+  class GetServerPathForTokenGenerationResponse {
+    private String serverUrl;
+    private  String errorMessage;
+
+    public GetServerPathForTokenGenerationResponse(String serverVersion, String errorMessage) {
```
Comment (sophio-japharidze-sonarsource) [inline src/main/java/org/sonarsource/sonarlint/ls/SonarLintExtendedLanguageServer.java:222]: Why is the first parameter called `serverVersion`?

**Review by sophio-japharidze-sonarsource — APPROVED** [review]

File: src/main/java/org/sonarsource/sonarlint/ls/SonarLintExtendedLanguageServer.java [hunk 2]
```diff
@@ -214,4 +214,50 @@ public void setConnectionId(String connectionId) {
...
+      return errorMessage;
+    }
+
+    public void setErrorMessage(String errorMessage) {
+      this.errorMessage = errorMessage;
+    }
+
+  }
+
+  class GetServerPathForTokenGenerationParams {
```
Comment (jblievremont) [inline src/main/java/org/sonarsource/sonarlint/ls/SonarLintExtendedLanguageServer.java:245]: Is this class needed? I believe it should be possible to pass a `String` directly (and avoid the boilerplate 😛)

**Review by jblievremont — APPROVED** [review]
Minor feedback on the LSP signature, else LGTM 👍🏻

File: src/main/java/org/sonarsource/sonarlint/ls/SonarLintExtendedLanguageServer.java (hunk 2 above)
Comment (kirill-knize-sonarsource) [reply src/main/java/org/sonarsource/sonarlint/ls/SonarLintExtendedLanguageServer.java:245]: As discussed, this is the pattern we use for this parameters and it's for the consistency of the code style. We can decide later if we want to refactor it.

**Pull request merged**